pkg/controllers/updaterun/controller.go

```diff
@@ -79,6 +79,12 @@ func (r *Reconciler) Reconcile(ctx context.Context, req runtime.Request) (runtim
 	}
 	runObjRef := klog.KObj(&updateRun)
 
+	// Remove waitTime from the updateRun status for AfterStageTask for type Approval.
```

Comment:
IIRC, this is just to fix the existing approval CRs?

Reply:
why put it here instead of in the init?

Reply:
Fix existing updateRun CRs. We have to do it here because when an updateRun is executing or being deleted, the reconciler won't enter initialization any more. In deletion, we update to remove the finalizer; during execution/validation, we update the updateRun status. All these updates will fail if the `waitTime` is not set to nil.

Reply:
discussed with Arvind, looks like it's enough to just correct the in-memory status and piggyback that when the status is updated at the end of the reconcile.